File: examples/hello.py
#!/usr/bin/env python
import sys
from gemmi import cif

greeted = set()
for path in sys.argv[1:]:
    try:
        doc = cif.read_file(path)  # copy all the data from mmCIF file
        block = doc.sole_block()  # mmCIF has exactly one block
        for element in block.find_loop("_atom_site.type_symbol"):
            if element not in greeted:
                print("Hello " + element)
                greeted.add(element)
    except Exception as e:
        print("Oops. %s" % e)
        sys.exit(1)

File: examples/hello.cc
#include <iostream>
#include <set>
#include <gemmi/cif.hpp>

namespace cif = gemmi::cif;

int main(int argc, char* argv[]) {
  std::set<std::string> greeted;
  if (argc != 2) return 1;
  try {
    cif::Document doc = cif::read_file(argv[1]);
    cif::Block& block = doc.sole_block(); // mmCIF has exactly one block
    for (const std::string &s : block.find_loop("_atom_site.type_symbol"))
      if (greeted.insert(s).second) // insert() returns pair<iterator,bool>
        std::cout << "Hello " << s << std::endl;
  } catch (std::exception& e) {
    std::cerr << "Oops. " << e.what() << std::endl;
    return 1;
  }
  return 0;
}
